[quant] Added relu6 kernel

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#24799 [quant] Added relu6 kernel**

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24799

Differential Revision: [D16875493](https://our.internmc.facebook.com/intern/diff/D16875493)

Commit:
Added relu6 kernel

Commit:
Update on "Added relu6 kernel"


Differential Revision: [D16875493](https://our.internmc.facebook.com/intern/diff/D16875493)

Commit:
Update on "Added relu6 kernel"


Differential Revision: [D16875493](https://our.internmc.facebook.com/intern/diff/D16875493)

jerryzh168 commented on torch/nn/quantized/modules/activation.py:
```diff
@@ -39,3 +41,38 @@ def forward(self, input):
         return ReLU(mod.inplace)
+
+
+class ReLU6(NNReLU):
```
nit: let's use nn.ReLU to be more explict. same for the previous one.

jerryzh168 commented on aten/src/ATen/native/quantized/cpu/qrelu.cpp:
```diff
@@ -3,6 +3,7 @@
 #include <ATen/core/op_registration/op_registration.h>
 #include <ATen/native/TensorIterator.h>
 #include <ATen/native/cpu/Loops.h>
+#include <ATen/quantized/Quantizer.h>
```
unused?

zafartahirov replied:
I think I need it for the `at::quantize_val`.

jerryzh168 reviewed (approved):
LG, some nit comments

llyfacebook commented on torch/nn/quantized/modules/activation.py:
```diff
@@ -39,3 +41,38 @@ def forward(self, input):
+    """
+    def __init__(self, inplace=False):
+        super(ReLU6, self).__init__(inplace)
+        assert not inplace, 'torch.nn.quantized.ReLU does not support inplace'
```
Need inplace, :)

zafartahirov replied:
Inplace is part of another diff

llyfacebook commented on torch/nn/quantized/modules/activation.py:
```diff
@@ -39,3 +41,38 @@ def forward(self, input):
+
+    @staticmethod
+    def from_float(mod):
+        return ReLU(mod.inplace)
```
ReLU6(mod.inplace)?

Commit:
Update on "Added relu6 kernel"


Differential Revision: [D16875493](https://our.internmc.facebook.com/intern/diff/D16875493)

Commit:
Update on "Added relu6 kernel"


Differential Revision: [D16875493](https://our.internmc.facebook.com/intern/diff/D16875493)

Commit:
Update on "Added relu6 kernel"


Differential Revision: [D16875493](https://our.internmc.facebook.com/intern/diff/D16875493)

Commit:
Update on "[quant] Added relu6 kernel"


Pull Request resolved: https://github.com/pytorch/pytorch/pull/24799

Differential Revision: [D16875493](https://our.internmc.facebook.com/intern/diff/D16875493)
